src/sagemaker/serve/builder/requirements_manager.py

```diff
@@ -0,0 +1,98 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+"""Requirements Manager class to pull in client dependencies from a .txt or .yml file"""
+from __future__ import absolute_import
+import logging
+import os
+import subprocess
+
+logger = logging.getLogger(__name__)
+
+
+class RequirementsManager:
+    """Transformers build logic with ModelBuilder()"""
+
+    def detect_file_exists(self, dependencies: str = None) -> str:
+        """Creates snapshot of the user's environment
```

Comment:
Nit: Not sure if this is entirely right? we are detecting not snapshotting anything?

Reply:
Yes that is right, good catch, I will rephrase.